packages/itwinui-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -82,12 +82,25 @@ export type ComboboxMultipleTypeProps<T> =
        * Callback fired when selected value changes.
        */
       onChange?: (value: T) => void;
+      /**
+       * Only applicable when `multiple` is enabled.
+       *
+       * If `true`, toggling an option will clear the filter.
+       * Useful when users would likely want to re-filter after toggling an option.
+       *
+       * If `false`, the filter will remain as-is after toggling an option.
+       * Useful when users would likely want to toggle multiple options from the _same_ filtered results.
+       *
+       * @default true
+       */
+      clearFilterOnOptionToggle?: never;
```

Comment:
I'm quite uncertain about the type `never` being used here instead of `boolean`. Would you mind elaborating a little more on this?

Reply:
I believe this TS pattern is referred to as "discriminated unions".
- https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions
- https://basarat.gitbook.io/typescript/type-system/discriminated-unions
- https://www.totaltypescript.com/discriminated-unions-are-a-devs-best-friend

Reply:
This is really helpful. Thank you!